### PR TITLE
Make UDPConnection Listenable

### PR DESCRIPTION
**Pull request by rowleya**
Makes a UDPConnection usable with a ConnectionListener

### CODE REVIEW

File: spinnman/connections/udp_packet_connections/udp_connection.py [hunk 1]
```diff
@@ -242,3 +243,6 @@ def __repr__(self):
         return _REPR_TEMPLATE.format(
             self.local_ip_address, self.local_port,
             self.remote_ip_address, self.remote_port)
+
+    def get_receive_method(self):
```
Comment (dkfellows) [inline spinnman/connections/udp_packet_connections/udp_connection.py:247]: Missing an `@override` there.

Comment (rowleya) [reply spinnman/connections/udp_packet_connections/udp_connection.py:248]: Oh yes, added now

**Review by dkfellows — APPROVED** [review]
If everything that is a `Connection` is also a `Listenable` (because `UDPConnection` is the concrete base class for them all IIRC) then is there any need for `Listenable` at all? 🤔 

The change is good as it is.

**Comment by rowleya** [comment]
> If everything that is a `Connection` is also a `Listenable` (because `UDPConnection` is the concrete base class for them all IIRC) then is there any need for `Listenable` at all? 🤔
> 
> The change is good as it is.

Yes, looks like something to look at in a clean up!

**Pull request merged**